streams/src/main/java/org/apache/kafka/streams/kstream/internals/foreignkeyjoin/SubscriptionResponseWrapper.java

```diff
@@ -21,22 +21,30 @@
 import java.util.Arrays;
 
 public class SubscriptionResponseWrapper<FV> {
-    final static byte CURRENT_VERSION = 0x00;
+    final static byte CURRENT_VERSION = 0x01;
+    // 0x00
     private final long[] originalValueHash;
     private final FV foreignValue;
     private final byte version;
+    // 0x01
+    private final Integer primaryPartition;
```

Comment:
While we need this field in this class, so we actually need to serialize it? It seems it must not go over the wire, does it?

Thus, for the subscription response, we don't need the dump the version? -- We only need it on the right hand side to write the response into the topic, but we can do this in-memory only?

If this is correct, we also don't need to update the corresponding Serde.

Reply:
Good catch! Done in f7f9181ec3312b21a3534919819ccd02dfdcbc74.